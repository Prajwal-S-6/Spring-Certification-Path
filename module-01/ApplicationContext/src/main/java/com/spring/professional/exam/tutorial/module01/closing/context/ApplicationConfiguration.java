package com.spring.professional.exam.tutorial.module01.closing.context;

import org.springframework.context.annotation.ComponentScan;

@ComponentScan
public class ApplicationConfiguration {
}
