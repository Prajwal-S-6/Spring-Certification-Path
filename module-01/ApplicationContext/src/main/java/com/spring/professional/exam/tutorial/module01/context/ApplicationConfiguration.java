package com.spring.professional.exam.tutorial.module01.context;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

@Configuration
@ComponentScan
public class ApplicationConfiguration {

    @Bean
    String someBean() {
        return "Test";
    }
}
