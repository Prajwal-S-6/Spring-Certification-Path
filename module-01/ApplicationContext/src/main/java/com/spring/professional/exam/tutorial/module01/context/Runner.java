package com.spring.professional.exam.tutorial.module01.context;

import com.spring.professional.exam.tutorial.module01.context.beans.Car;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.context.annotation.ComponentScan;

@ComponentScan
public class Runner {
    public static void main(String... args) {
        try (AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(ApplicationConfiguration.class)) {
            Car car = context.getBean(Car.class);
            // System.out.println(context.getBean(String.class));
            car.drive();
        }
    }
}
