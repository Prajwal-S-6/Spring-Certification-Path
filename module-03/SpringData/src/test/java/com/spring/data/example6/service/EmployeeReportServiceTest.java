package com.spring.data.example6.service;


import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.system.CapturedOutput;
import org.springframework.boot.test.system.OutputCaptureExtension;


@SpringBootTest
@ExtendWith(OutputCaptureExtension.class)
class EmployeeReportServiceTest {

    @Autowired
    private EmployeeReportService employeeReportService;

    @BeforeAll
    static void setUp() {
        System.out.close();
    }

    @Test
    void shouldCreateNewConnectionForEachMethod(CapturedOutput capturedOutput) {
        employeeReportService.printReport();
        Approvals.verify(capturedOutput.getOut().lines().filter(string -> string.contains("getConnection")).reduce((val1, val2) -> val1 + "\n" + val2).get());
    }

    @Test
    void shouldCreateOneConnectionForAllMethods(CapturedOutput capturedOutput) {
        employeeReportService.printReportInTransaction();
        Approvals.verify(capturedOutput.getOut().lines().filter(string -> string.contains("getConnection")).reduce((val1, val2) -> val1 + "\n" + val2).get());
    }
}