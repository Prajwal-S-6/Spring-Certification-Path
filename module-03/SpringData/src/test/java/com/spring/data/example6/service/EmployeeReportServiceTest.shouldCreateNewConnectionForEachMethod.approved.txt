Data Source Trace: Connection com.zaxxer.hikari.HikariDataSource.getConnection()
Data Source Trace: Connection com.zaxxer.hikari.HikariDataSource.getConnection()
Data Source Trace: Connection com.zaxxer.hikari.HikariDataSource.getConnection()
Data Source Trace: Connection com.zaxxer.hikari.HikariDataSource.getConnection()
Data Source Trace: Connection com.zaxxer.hikari.HikariDataSource.getConnection()